src/main/java/com/timgroup/statsd/UnixStreamClientChannel.java

```diff
@@ -136,9 +136,11 @@ private void connect() throws IOException {
         }
         if (!delegate.connect(address)) {
             if (connectionTimeout > 0 && System.nanoTime() > deadline) {
+                delegate.close();
```

Comment:
Would it make sense to do this with a `try/finally` block? AFAICS `setOption` can throw too, and we'd leak then too.

Reply:
not sure how you would use a finally here 🤔 but I tried a `try`

Reply:
You are right, `catch` makes more sense of course.